lib/src/features/city_details/data/data_sources/remote_data_source.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:emotion/src/features/city_details/data/models/city_details_model.dart';
+import 'package:http/http.dart';
+
+abstract class CityDetailsRemoteDataSource {
+  Future<CityDetailsModel> getCityDetails(String cityName, String countryName);
+}
+
+class CityDetailsRemoteDataSourceImpl extends CityDetailsRemoteDataSource {
+  final http.Client client;
+
+  CityDetailsRemoteDataSourceImpl({required this.client});
+
+  @override
+  Future<CityDetailsModel> getCityDetails(String cityName, String countryName) async {
+    final response = await client.get(
+      Uri.parse('https://dev.escooters.blumil.pl/api/$countryName/$cityName')
```

Comment:
Also, I would prefer storing the base URL in a configuration file or a .env file. This way, if you need to build the app for production or different environments, you won't have to manually update every instance of the API URL in the application code.

I like using the [flutter_dotenv](https://pub.dev/packages/flutter_dotenv) package